addons/project/wizard/project_share_wizard.py

```diff
@@ -61,13 +85,44 @@ def action_share_record(self):
             'context': self.env.context,
         }
 
+    def _send_link(self):
+        template_vals = [{'partner_id': partner.id, 'project_id': self.resource_ref.id, 'is_portal': partner.user_ids} for partner in self.partner_ids]
+        templates = self.env['project.share.wizard.template'].create(template_vals)
```

Comment:
It is sadly not possible. If we use the project, then only one template will be correctly loaded. instead of having one template for each partners, we're going to generate a mail for the first partner in the many2one relation of the project and the others will be ignored. I've discussed with the team in charge of the templates, and sadly, the creation of a new model is mandatory. Since one distinct instance is needed per mail created.